src/main/kotlin/com/exactpro/th2/codec/AvroCodec.kt

```diff
@@ -85,12 +100,12 @@ class AvroCodec(
                 )
             }
             val schemaId = byteBuf.readInt()
-            reader = checkNotNull(datumReaders[schemaId]) { "No reader found for schema id: $schemaId" }
+            reader = schemaIdResolver!!.getReader(schemaId)
             decoder = DecoderFactory.get().binaryDecoder(bytes, AVRO_HEADER_SIZE, bytes.size - AVRO_HEADER_SIZE, null)
             id = schemaId
         } else {
             check(sessionAlias.isNotEmpty()) {"Session alias cannot be empty. Raw message: $rawMessage"}
-            reader = checkNotNull(datumReaders[sessionAlias]) { "No reader found for session alias: $sessionAlias" }
+            reader = aliasResolver!!.getReader(sessionAlias)
```

Comment:
Maybe split via strategy pattern or any abstraction. `!!` operator is bad practice

Reply:
!! operator is not a bad practice if we know what we are doing.